Sources/Plasma/FeatureLib/pfPython/cyMisc.cpp

```diff
@@ -504,66 +504,44 @@ ST::string cyMisc::GetClientName(pyKey &avKey)
 
 PyObject* cyMisc::GetAvatarKeyFromClientID(int clientID)
 {
-    PyObject* keyObj = nullptr;
-
     if (clientID == plNetClientMgr::GetInstance()->GetPlayerID())
     {
-        keyObj = pyKey::New(plNetClientMgr::GetInstance()->GetLocalPlayerKey());
+        return pyKey::New(plNetClientMgr::GetInstance()->GetLocalPlayerKey());
     }
     else
     {
-        plNetTransportMember **members = nullptr;
-        plNetClientMgr::GetInstance()->TransportMgr().GetMemberListDistSorted( members );
+        std::vector<plNetTransportMember*> members = plNetClientMgr::GetInstance()->TransportMgr().GetMemberListDistSorted();
 
-        if (members != nullptr)
+        for (plNetTransportMember* mbr : members)
         {
-            for(int i = 0; i < plNetClientMgr::GetInstance()->TransportMgr().GetNumMembers(); i++ )
+            if (mbr != nullptr && mbr->GetAvatarKey() != nullptr && mbr->GetPlayerID() == clientID)
             {
-                plNetTransportMember *mbr = members[ i ];
-                if (mbr != nullptr && mbr->GetAvatarKey() != nullptr && mbr->GetPlayerID() == clientID)
-                {   
-                    keyObj = pyKey::New(mbr->GetAvatarKey());
-                    break;
-                }
+                return pyKey::New(mbr->GetAvatarKey());
             }
         }
-
-        delete [] members;
     }
 
-    if (keyObj)
-        return keyObj;
-    else
-        PYTHON_RETURN_NONE;
+    PYTHON_RETURN_NONE;
 }
 
 
 int cyMisc::GetClientIDFromAvatarKey(pyKey& avatar)
 {
-    int ret = -1;
-
     if (plNetClientMgr::GetInstance()->GetLocalPlayerKey() == avatar.getKey())
     {
         return (plNetClientMgr::GetInstance()->GetPlayerID());
     }
-    plNetTransportMember **members = nullptr;
-    plNetClientMgr::GetInstance()->TransportMgr().GetMemberListDistSorted( members );
-    if (members != nullptr)
+
+    std::vector<plNetTransportMember*> members = plNetClientMgr::GetInstance()->TransportMgr().GetMemberListDistSorted();
```

Comment:
Does this need a sorted list? Surely there is a way to get a plain old list or forward this request to some other code..